sched/pthread/pthread_condclockwait.c

```diff
@@ -202,134 +128,64 @@ int pthread_cond_clockwait(FAR pthread_cond_t *cond,
-#endif
-              /* Give up the mutex */
 
-              mutex->pid = INVALID_PROCESS_ID;
-#ifndef CONFIG_PTHREAD_MUTEX_UNSAFE
-              mflags     = mutex->flags;
-#endif
-#ifdef CONFIG_PTHREAD_MUTEX_TYPES
-              type       = mutex->type;
-              nlocks     = mutex->nlocks;
-#endif
-              ret        = pthread_mutex_give(mutex);
-              if (ret != 0)
-                {
-                  /* Restore interrupts  (pre-emption will be enabled
-                   * when we fall through the if/then/else)
-                   */
-
-                  leave_critical_section(flags);
-                }
-              else
-                {
-                  /* Start the watchdog */
-
-                  wd_start(&rtcb->waitdog, ticks,
-                           pthread_condtimedout, mypid);
-
-                  /* Take the condition semaphore.  Do not restore
-                   * interrupts until we return from the wait.  This is
-                   * necessary to make sure that the watchdog timer and
-                   * the condition wait are started atomically.
-                   */
-
-                  status = nxsem_wait((FAR sem_t *)&cond->sem);
-
-                  /* We no longer need the watchdog */
-
-                  wd_cancel(&rtcb->waitdog);
-
-                  /* Did we get the condition semaphore. */
-
-                  if (status < 0)
-                    {
-                      /* NO.. Handle the special case where the semaphore
-                       * wait was awakened by the receipt of a signal --
-                       * presumably the signal posted by
-                       * pthread_condtimedout().
-                       */
-
-                      if (status == -EINTR)
-                        {
-                          swarn("WARNING: Timedout!\n");
-                          ret = ETIMEDOUT;
-                        }
-                      else
-                        {
-                          ret = status;
-                        }
-                    }
-
-                  /* The interrupts stay disabled until after we sample
-                   * the errno.  This is because when debug is enabled
-                   * and the console is used for debug output, then the
-                   * errno can be altered by interrupt handling! (bad)
-                   */
-
-                  leave_critical_section(flags);
-                }
-
-              /* Reacquire the mutex (retaining the ret). */
-
-              sinfo("Re-locking...\n");
-
-              status = pthread_mutex_take(mutex, NULL, false);
-              if (status == OK)
-                {
-                  mutex->pid    = mypid;
+          /* The interrupts stay disabled until after we sample
+           * the errno.  This is because when debug is enabled
+           * and the console is used for debug output, then the
+           * errno can be altered by interrupt handling! (bad)
+           */
```

Comment:
I do not see the code that corresponds to this description. In other words I do not see where `errno` is accessed here. I see that it was originally here, but do not understand if it is needed.

I see that critical section and preemption disable was used to protect `clock_abstime2ticks(clockid, abstime, &ticks);` call. Now it is removed, so maybe we need re-evaluate locking schema here.

I will re-examine changes more carefully later today and will provide additional reply.